### PR TITLE
Remove trailing translation delimiter = from titles

### LINKED ISSUE #702: Remove trailing translation delimiter (= equals sign) from titles
Translated titles of the form "lang 1 = lang 2" have been split, but the delimiter wasn't removed from the first language title. This affects both editions and the works derived from them.

An example is: https://openlibrary.org/books/OL20408332M/Cyllidebau_ysgolion

### LINKED ISSUE #3105: JSON Import: Subtitle splitting
This has been fixed for MARC and Amazon imports, but the json API currently appears to bypass this code. Correct subtitle splitting is necessary for the current system’s existing work matching algorithm. Inconsistently split titles create easily avoidable duplicate works and editions.

### Additional context

After investigation:
MARC records store subtitle separately , 245$a for main title, and 245$b for 'remainder' (=subtitle)
The expected test data shows how this is processed into JSON for importing by the MARC code:
https://github.com/internetarchive/openlibrary/tree/master/openlibrary/catalog/marc/tests/test_data/bin_expect
and 
https://github.com/internetarchive/openlibrary/tree/master/openlibrary/catalog/marc/tests/test_data/xml_expect

with sufficient examples.

The Amazon path uses a custom title splitter method in `vendors.py` 
https://github.com/internetarchive/openlibrary/blob/f9354ee93bb3d89b2339f2209a319d28104ea7e6/openlibrary/core/vendors.py#L255-L271

Which could / should be reused in the main `catalog.add_book.load()` path iff a subtitle does not already exist and there is an ISBD subtitle separator in the current `title` data field.

### PR DESCRIPTION
**Pull request by hornc**
Closes #702 
This has become prep-work for #3105 since it invloves investigating and refactoring how existing imported titles are handled.

### Technical

Builds on #3106 which should be merged first. **DONE!**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
         # On pull requests, run all flake8 tests on modified code
         # if clause avoids `fatal: ambiguous argument 'origin/master'` on git push
         - if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
-            git diff origin/master -U0 | flake8 --diff --max-line-length=88;
+            git diff origin/master -U0 | flake8 --diff --max-line-length=127;
```
Comment (cclauss) [inline .travis.yml]: Let’s stick with 88 characters max per line. That is reasonable and will align with psf/black.

File: openlibrary/catalog/add_book/__init__.py [hunk 2]
```diff
@@ -214,11 +215,11 @@ def add_cover(cover_url, ekey, account=None):
     :rtype: int or None
     :return: Cover id, or None if upload did not succeed
     """
-    olid = ekey.split("/")[-1]
+    olid = ekey.split('/')[-1]
     coverstore_url = config.get('coverstore_url').rstrip('/')
     upload_url = coverstore_url + '/b/upload2'
     if upload_url.startswith("//"):
-        upload_url = "{0}:{1}".format(web.ctx.get("protocol", "http"), upload_url)
+        upload_url = "{0}:{1}".format(web.ctx.get('protocol', 'http'), upload_url)
```
Comment (cclauss) [inline openlibrary/catalog/add_book/__init__.py]: Why not change the first four double quotes to be like the last four single quotes?

Comment (hornc) [reply openlibrary/catalog/add_book/__init__.py]: Thanks @cclauss , fixed.

**Comment by hornc** [comment]
@cclauss I have reverted the line length change and made this code compatible with the shorter length. Ready for review.

**Review by cclauss — APPROVED** [review]

**Comment by mekarpeles** [comment]
Moving forward can we please make e.g. quote changes in a separate PR (possibly in bulk)? This seems like something we may want to add specifically to our `git-diff` linter?

**Pull request merged**